--- apache-log4cxx-0.10.0/configure	2008-04-01 00:44:09.000000000 +0200
+++ apache-log4cxx-0.10.0/configure	2009-12-02 19:55:26.000000000 +0100
@@ -1693,7 +1693,7 @@
     # OSF1 and SCO ODT 3.0 have their own names for install.
     # Don't use installbsd from OSF since it installs stuff as root
     # by default.
-    for ac_prog in ginstall scoinst install; do
+    for ac_prog in install scoinst ginstall; do
       for ac_exec_ext in '' $ac_executable_extensions; do
 	if $as_executable_p "$as_dir/$ac_prog$ac_exec_ext"; then
 	  if test $ac_prog = install &&
@@ -1705,7 +1705,7 @@
 	    # program-specific install script used by HP pwplus--don't use.
 	    :
 	  else
-	    ac_cv_path_install="$as_dir/$ac_prog$ac_exec_ext -c"
+	    ac_cv_path_install="/usr/bin/install -c"
 	    break 3
 	  fi
 	fi
